internal/app/referenceclient/tracer.go

```diff
@@ -117,53 +118,59 @@ func buildWireDetails(ctx context.Context) (*v1.WireDetails, error) {
 	if trace == nil || trace.Response == nil {
 		return nil, nil //nolint:nilnil
 	}
+
 	statusCode := int32(trace.Response.StatusCode)
+	details := &v1.WireDetails{
+		ActualStatusCode:   statusCode,
+		ActualHttpTrailers: internal.ConvertToProtoHeader(trace.Response.Trailer),
+	}
 
-	var jsonRaw structpb.Struct
 	contentType := trace.Response.Header.Get("content-type")
-
-	// If this is a unary request that returned an error, then use the entire
-	// response body as the wire error details.
-	if isUnaryJSONError(contentType, statusCode) { //nolint:nestif
+	switch {
+	case isUnaryJSONError(contentType, statusCode):
+		// If this is a unary request that returned an error, then use the entire
+		// response body as the wire error details.
 		body, err := io.ReadAll(wrapper.buf)
 		if err != nil {
 			return nil, err
 		}
-		if err := protojson.Unmarshal(body, &jsonRaw); err != nil {
+		details.ConnectErrorRaw = &structpb.Struct{}
+		if err := protojson.Unmarshal(body, details.ConnectErrorRaw); err != nil {
 			return nil, err
 		}
-	} else if strings.HasPrefix(contentType, "application/connect+") {
-		// If this is a streaming request, then look through the trace events
+	case strings.HasPrefix(contentType, "application/connect+"):
+		// If this is a streaming Connect request, then look through the trace events
 		// for the ResponseBodyEndStream event and parse its content into an
 		// endStreamError to see if there are any error details.
-		type endStreamError struct {
+		endStreamContent, ok := getBodyEndStream(trace)
+		if !ok {
+			break
```

Comment:
I was originally thinking we should report an error here if no `EndStream` message is found for a streaming response. But I guess that would be caught by a test failure with a missing `ConnectErrorRaw` in the response?

Reply:
Correct. Also, we don't expect one for things like cancellations or timeouts. So we'd have to embed way too much logic to get the check right, and the reference client, by virtue of implementing the RPC protocols, is already enforcing the things that need to be enforced. This code is just to scrape add'l on-the-wire details so we can do add'l validation (and be a little more strict than connect-go).